packages/core/src/odata-common/selectable/selectable.ts

```diff
@@ -29,3 +23,19 @@ export type Selectable<EntityT extends Entity> =
         | CollectionField<EntityT>
         | AllFields<EntityT>
     : never;
+
+/**
+ * Interface to denote a selectable [[Field]].
+ */
+export interface SelectableField {
```

Comment:
Question: The property selectable feels like a boolean like the `Nullable` which is a property of the field. Could we include this also in the generic signature of the field?

Reply:
I have thought about this 3mins so this is just a basis for discussion.

Reply:
As discussed.